startupscript/dataproc/install-r.sh

```diff
@@ -0,0 +1,127 @@
+# This is intentionally not a Bash alias as they are not supported in shell scripts.
+readonly RUN_AS_LOGIN_USER="sudo -u ${LOGIN_USER} bash -l -c"
+
+# Define directories and paths to binaries and system file
+readonly USER_HOME_DIR="/home/${LOGIN_USER}"
+readonly USER_WORKBENCH_CONFIG_DIR="${USER_HOME_DIR}/.workbench"
+readonly USER_BASHRC="${USER_HOME_DIR}/.bashrc"
+
+readonly OUTPUT_FILE="${USER_WORKBENCH_CONFIG_DIR}/install-r-output.txt"
+
+readonly CONDA_DIR='/opt/conda/miniconda3'
+readonly CONDA_BIN_DIR="${CONDA_DIR}/bin"
+readonly R_HOME_DIR='/usr/lib/R'
+readonly R_BIN_DIR="${R_HOME_DIR}/bin"
+readonly RUN_R="${R_BIN_DIR}/R"
+
+# Split stdout and stderr in the rest of this script to an output file for debugging.
+# But still output to stdout and stderr so users can also debug via the initialization
+# action output files in the cluster staging bucket.
+exec > >(tee -a "${OUTPUT_FILE}") 2>&1
+exec 2>&1
+
+emit "Installing R ..."
+
+# Ensure that the user's workbench configuration directory exists
+mkdir -p "${USER_WORKBENCH_CONFIG_DIR}"
+${RUN_AS_LOGIN_USER} "mkdir -p '${USER_WORKBENCH_CONFIG_DIR}'"
+
+# Add CRAN R archive network repository
+add-apt-repository "deb https://cloud.r-project.org/bin/linux/debian $(lsb_release -cs)-cran40/"
+
+# Fetch and export the repository's gpg key
+# See debian package installation instructions:
+# https://cran.r-project.org/bin/linux/debian/
+gpg --keyserver keyserver.ubuntu.com \
+  --recv-key '95C0FAF38DB3CCAD0C080A7BDC78B2DDEABC47B7'
+gpg --armor --export '95C0FAF38DB3CCAD0C080A7BDC78B2DDEABC47B7' |
+  sudo tee /etc/apt/trusted.gpg.d/cran_debian_key.asc
+
+# Install r-base package
+apt-get update -y
+apt-get install r-base -y
+
+# Add R to the PATH variable in user's bashrc
+cat <<EOF >>"${USER_BASHRC}"
+# Prepend "${R_BIN_DIR}" (if not already in the path)
+if [[ ":\${PATH}:" != *":${R_BIN_DIR}:"* ]]; then
+  export PATH="${R_BIN_DIR}":"\${PATH}"
+fi
+EOF
+
+# Install IRKernel on the master node to support interactive R notebooks in Jupyter
+# Note: We set the PATH to ensure that R knows where to find the 'jupyter' binary
+# See IR kernel installation docs:
+# https://irkernel.github.io/installation/
+if [[ "${DATAPROC_NODE_ROLE}" == 'Master' ]]; then
+  PATH="${CONDA_BIN_DIR}:${PATH}" "${RUN_R}" -e "\
+  install.packages('IRkernel', repos='http://cran.rstudio.com/');
+  IRkernel::installspec()"
+
+  # Point IR jupyter kernel to the newly installed R
+  readonly IR_KERNEL="${CONDA_DIR}/share/jupyter/kernels/ir/kernel.json"
+  cat <<EOF >"${IR_KERNEL}"
+{
+  "argv": [
+    "${RUN_R}",
+    "--slave",
+    "-e",
+    "IRkernel::main()",
+    "--args",
+    "{connection_file}"
+  ],
+  "display_name": "R",
+  "language": "R",
+  "env": {
+    "R_HOME": "${R_HOME_DIR}",
+    "SPARK_HOME": "/usr/lib/spark"
+  }
+}
+EOF
+
+  # Restart jupyter service to pick up the new R kernel
+  readonly JUPYTER_SERVICE_NAME='jupyter.service'
```

Comment:
This feels like a strange place to declare a readonly variable. Any reason not to declare it up around the other R variables (lines 55-57), like in [post-startup.sh](https://github.com/verily-src/workbench-app-devcontainers/blob/772786274b0d32d8748982f5c6dc7e82891906dc/startupscript/dataproc/startup.sh#L98)?